pkg/app/piped/eventwatcher/eventwatcher.go

```diff
@@ -165,52 +176,49 @@ func (w *watcher) run(ctx context.Context, repo git.Repo, repoCfg config.PipedRe
 	}
 }
 
+// cloneRepo clones the git repository under the working directory.
+func (w *watcher) cloneRepo(ctx context.Context, repoCfg config.PipedRepository) (git.Repo, error) {
+	repo, err := w.gitClient.Clone(ctx, repoCfg.RepoID, repoCfg.Remote, repoCfg.Branch, filepath.Join(w.workingDir, repoCfg.RepoID))
+	if err != nil {
+		w.logger.Error("failed to clone repository",
+			zap.String("repo-id", repoCfg.RepoID),
+			zap.Error(err),
+		)
+		return nil, fmt.Errorf("failed to clone repository %s: %w", repoCfg.RepoID, err)
+	}
+	return repo, nil
+}
+
 // updateValues inspects all Event-definition and pushes the changes to git repo if there is.
 func (w *watcher) updateValues(ctx context.Context, repo git.Repo, events []config.EventWatcherEvent, commitMsg string) error {
-	// Copy the repo to another directory to avoid pull failure in the future.
-	tmpDir, err := ioutil.TempDir("", "event-watcher")
+	// Copy the repo to another directory to modify local file to avoid reverting previous changes.
+	tmpDir, err := ioutil.TempDir(w.workingDir, "repo")
 	if err != nil {
 		return fmt.Errorf("failed to create a new temporary directory: %w", err)
 	}
-	defer os.RemoveAll(tmpDir)
 	tmpRepo, err := repo.Copy(filepath.Join(tmpDir, "tmp-repo"))
 	if err != nil {
 		return fmt.Errorf("failed to copy the repository to the temporary directory: %w", err)
 	}
+	defer tmpRepo.Clean()
 
-	commits := make([]*commit, 0)
 	for _, e := range events {
-		latestEvent, ok := w.eventGetter.GetLatest(ctx, e.Name, e.Labels)
-		if !ok {
+		if err := w.commitFiles(ctx, &e, tmpRepo, commitMsg); err != nil {
```

Comment:
`&e` is pointing to the same one. So we should passing struct instead of a pointer.

Reply:
It is. Actually it works well at this logic. But we'd prefer copying it.